workshop/3-Experimentation/3.5-Amplitude-Performance-Metrics.ipynb

```diff
@@ -0,0 +1,461 @@
+    "\n",
+    "![](images/amplitude/conversion-funnel-pre-settings.png)\n",
+    "\n",
+    "Select the ProductViewed, ProductAdded, CheckoutStarted, and OrderCompleted events in that order.  Once that is complete, select the data dropdown on the lower right hand side of the screen, and select the Between tab. Then input a start date 60 days before today as a start and 30 days before today as an end date.  This report will show conversion data for the 30 days *prior* to personalization being deployed in the Retail Demo Store.  \n",
+    "\n",
+    "![](images/amplitude/conversion-funnel-pre-dates.png)\n",
+    "\n",
+    "Your funnel report will look something like this, though the exact conversion numbers and number of events will vary slightly from this screen shot:\n",
+    "\n",
+    "![](images/amplitude/conversion-funnel-pre-pers.png)\n",
+    "\n",
+    "Click on the lower part of the OrderCompleted bar graph. Note that the purchase conversion rate for the main funnel in the Retail Demo Store is 23.8%.  Not bad, but would we see an improvement in conversion if users were to receive personalized recommendations while they are interacting with the Retail Demo Store?\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Measuring the Conversion Effects of Personalized Product Recommendations\n",
+    "\n",
+    "In the first workshop, you deployed the following Personalization features in the Retail Demo Store:\n",
+    "\n",
+    "1. Personalized product recommendations\n",
+    "2. Similar product recommendations\n",
+    "3. Personalized search results re-ranking\n",
+    "\n",
+    "In order to measure how users convert after interacting with these features, we need to add a parameter to the event tracking sent to Amplitude, so that our funnel report can differentiate between product views and purchases that are *not* influenced via personalization features and product views and purchases that are a result of user interaction with the newly-added personalization features.\n",
+    "\n",
+    "In the `AnalyticsHandler.js` file, you will see code that adds a parameter to the ProductViewed event:\n",
+    "\n",
+    "```\n",
+    "if (this.amplitudeEnabled()) {\n",
+    "    // Amplitude event\n",
+    "    var eventProperties = {\n",
+    "        productId: product.id,\n",
+    "        name: product.name,\n",
+    "        category: product.category,\n",
+    "        image: product.image,\n",
+    "        feature: feature,\n",
+    "        experimentCorrelationId: experimentCorrelationId,\n",
+    "        price: +product.price.toFixed(2)\n",
+    "    };\n",
+    "    Amplitude.getInstance().logEvent('ProductViewed', eventProperties);\n",
+    "}\n",
+    "```\n",
+    "\n",
+    "The `feature` property will contain the name of the personalization feature that the user clicked on.  This will be used for personalized product recommendations and for similar product recommendations.  \n",
```

Comment:
The feature property represents the feature in the UI (home page recommendations, similar products, catalog details, search results) and is always sent with events regardless of whether the product clicked on came from a Personalize campaign or from default behavior. Does that mess things up?

If a Personalize campaign is being used to make the recommendations, the response from the Recommendations service includes an `X-Personalize-Recipe` response header with the recipe ARN for the campaign as the value. This is currently used to annotate the UI to let the demo user see that a campaign is in effect. Could this be used somehow as part of the event sent to Amplitude?

Reply:
I think this is a good thing actually - what I can do is add another route parameter that tracks the value of X-Personalize-Recipe, to behave like the experiment and the feature params.  This way this can be used to more easily distinguish between the different types of personalization in a given user event.

Reply:
Added this in a separate PR.